policy-test/src/lib.rs

```diff
@@ -349,6 +350,17 @@ pub fn endpoints_ready(obj: Option<&k8s::Endpoints>) -> bool {
     false
 }
 
+pub fn egress_network_traffic_policy_is(
+    policy: TrafficPolicy,
+) -> impl Condition<EgressNetwork> + 'static {
+    move |egress_net: Option<&EgressNetwork>| {
+        if let Some(egress_net) = &egress_net {
+            return egress_net.spec.traffic_policy == policy;
+        }
+        false
+    }
```

Comment:
Should this also include something like

```rust
egress_net
    .status
    .conditionsd
    .iter()
    .any(|condition| condition.type_ == "Accepted" && condition.status == "True")
```

to make sure the status controller has consumed it?

I notice there are some other places in tests, though, where we don't check statuses fully so maybe that's not necessary? E.g. [pub async fn await_grpc_route_status](https://github.com/linkerd/linkerd2/blob/d4bc37a979ca506ac079ad2fc450ef2d01282eff/policy-test/src/lib.rs#L273-L292)

Reply:
Good point, I will do one more pass over the tests and see whether there are opportunities to tighten things up in terms of status checking.

Reply:
I have added code to assert that egress network is accepted after creation or update in https://github.com/linkerd/linkerd2/pull/13390/commits/d99216a77652f65c8052370d8241f47a8acae16a

We can consider tightening up the assertions around route statuses as well as this is done sporadically now. I think however that this is beyond the cope of this PR as it spans most of the outbound API tests, not only the egress ones. It is a good change however.